src/layouts/dashboard/header/NotificationsPopover.js

```diff
@@ -20,77 +26,85 @@ import {
+    setNotifications(response.data.friendRequestList.map(makeNotifs));
+    function makeNotifs(obj) {
+    return {
+    id: obj.friendRequestID,
+    title: `${obj.userName} added you as a Friend!`,
+    description: `${obj.timeDiff} minutes ago`,
     avatar: null,
-    type: 'order_placed',
-    createdAt: set(new Date(), { hours: 10, minutes: 30 }),
-    isUnRead: true,
-  },
-  {
-    id: faker.datatype.uuid(),
-    title: faker.name.fullName(),
-    description: 'answered to your comment on the Minimal',
-    avatar: '/assets/images/avatars/avatar_2.jpg',
-    type: 'friend_interactive',
-    createdAt: sub(new Date(), { hours: 3, minutes: 30 }),
+    type: null,
+    createdAt: null,
     isUnRead: true,
-  },
-  {
-    id: faker.datatype.uuid(),
-    title: 'You have new message',
-    description: '5 unread messages',
-    avatar: null,
-    type: 'chat_message',
-    createdAt: sub(new Date(), { days: 1, hours: 3, minutes: 30 }),
-    isUnRead: false,
-  },
-  {
-    id: faker.datatype.uuid(),
-    title: 'You have new mail',
-    description: 'sent from Guido Padberg',
-    avatar: null,
-    type: 'mail',
-    createdAt: sub(new Date(), { days: 2, hours: 3, minutes: 30 }),
-    isUnRead: false,
-  },
-  {
-    id: faker.datatype.uuid(),
-    title: 'Delivery processing',
-    description: 'Your order is being shipped',
-    avatar: null,
-    type: 'order_shipped',
-    createdAt: sub(new Date(), { days: 3, hours: 3, minutes: 30 }),
-    isUnRead: false,
-  },
-];
+    name: obj.userName
+  }
+}
+  }).catch((err) => alert(err));
+}, 3000)
 
-export default function NotificationsPopover() {
-  const [notifications, setNotifications] = useState(NOTIFICATIONS);
+  const [openDialogue, setOpenDialogue] = useState(false);
+  const [tempFriendRequestID, setTempFriendRequestID] = useState(null);
+  const navigate = useNavigate();
+  const handleClickOpen = (notification) => {
+    setTempNotif(notification.name);
+    setTempFriendRequestID(notification.id);
+    setOpenDialogue(true);
+  };
+  const handleConfirm = () =>{
+    callWithToken('post', `${APIHOST}/api/friend/accept-request`, {friend_request_id: tempFriendRequestID}).then((response) => {
+    alert("Succesfully accepted friend request")
+    handleClose()
+    navigate('/', { replace: true});
+    }).catch((err) => {alert(err); handleClose()})
 
-  const totalUnRead = notifications.filter((item) => item.isUnRead === true).length;
+  }
+  const handleIgnore = () =>{
+    callWithToken('post', `${APIHOST}/api/friend/ignore-request`, {friend_request_id: tempFriendRequestID}).then((response) => {
+      alert("Succesfully accepted friend request")
+      handleClose()
+      navigate('/', { replace: true});
+      }).catch((err) => {alert(err); handleClose()})
+    
+    //  axios call to the api with tempFriendRequestID
+  }
+  const handleClose = () => {
+    setOpenDialogue(false);
+  };
 
+  const [notifications, setNotifications] = useState([]);
```

Comment:
states should be declared at the top of the function, right below "export default ..."